Frontend/road-management/src/views/TariffView/TariffWindow.tsx

```diff
@@ -0,0 +1,260 @@
+
+import { Fragment, useEffect, useState } from "react";
+import { Button, Dropdown } from "react-bootstrap";
+import DropdownButton from "react-bootstrap/DropdownButton";
+import Form from 'react-bootstrap/Form';
+import { ITariff, ITariffWindow, TariffDTO } from "../../interfaces/tariff/tariffinterfaces";
+import { TariffModelProxy } from "../../models/TariffModelProxy";
+
+
+const TariffWindow = () => {
+    const [editing, setEditing] = useState(false);
+    const [editingField, setEditingField] = useState(false);
+    const [addFormData, setAddFormData] = useState<TariffDTO | undefined>(
+        undefined
+    );
+    const [editFormData, setEditFormData] = useState<TariffDTO | undefined>(
+        undefined
+    );
+    const [infoMessage, setInfoMessage] = useState('');
+    const [tariffList, setTariffList] = useState<TariffDTO[]>(
+        new Array<TariffDTO>()
+    );
+    const [selectedTariff, setSelectedTariff] = useState<TariffDTO | undefined>(
+        undefined
+    );
+    const [newSelectedTariff, setNewSelectedTariff] = useState<TariffDTO | undefined>(
+        undefined
+    );
+
+    useEffect(() => {
+        const proxy = new TariffModelProxy();
+        async function fetchFromApi() {
+            setTariffList(await proxy.getTariffList());
+        }
+        fetchFromApi();
+    }, []);
+
+
+
+    class TariffDispatcher implements ITariffWindow {
+        showStatusMessage(result: boolean): void {
+            setInfoMessage(`${result ? `Udało się dodać taryfikator!` : 'Nie udało się dodać taryfikatora!'}`);
+        }
+    }
+
+    const tariffDispatcher = new TariffDispatcher();
+
+
+    class TariffPresenter implements ITariff {
+
+
+
+        onAddFormChanged = (event: any) => {
+            event.preventDefault();
+
+            const fieldName = event.target.getAttribute("name");
+            const fieldValue = event.target.value;
+
+
+            const newFormData = { ...addFormData };
+            newFormData[fieldName] = fieldValue;
+            if(selectedTariff !== undefined) {
+                new Map(Object.entries(selectedTariff!.rates)).set(fieldName, fieldValue)
+            setNewSelectedTariff(selectedTariff);
+        }
+
+        };
+
+        sendTariffChangeDataToSave = async (event: any) => {
+            event.preventDefault();
+
+            const fieldName = event.target.getAttribute("name");
+            const fieldValue = event.target.value;
+
+            if(selectedTariff !== undefined){
+            const tariffProxy = new TariffModelProxy();
+                
+            selectedTariff.rates = new Map(Object.entries(selectedTariff!.rates)).set(fieldName, +fieldValue);
+
+            let proxyResponse = await tariffProxy.updateTariff(selectedTariff);
+        }
+        };
+
+        sendNewTariffDataToSave = async (event: any) => {
+            event.preventDefault();
+            if (selectedTariff !== undefined) {
+                const newtariff = {
```

Comment:
To powinno być dodawanie nowego taryfikatora do konkretnych dróg. Natomiast to jest dodawanie nowego rate do rates taryfikatora. Jest to jakaś interpretacja "Dodania taryfikatora", ale raczej nie o to chodziło.

Reply:
(Rates to to): 
![image](https://user-images.githubusercontent.com/56975218/207132370-03b39bd3-0ad9-468f-bd47-3ce57705d30f.png)